### PR TITLE
Clown Mutation (Clumsy) and Abnormal Mutations can no longer be cured.

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

clown cant be cured from clusmy
fixes oversight in mutadone proof not working

## Why It's Good For The Game

clown that cures himself from clumsy is cringe

## Changelog
:cl:
fix: clowns cannot cure their clumsyness with mutadone anymore
fix: fixes mutadone proofness not working
balance: mutadone no longer cures "weird" mutations such as those you get from the wizard
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/clown.dm [hunk 1]
```diff
@@ -57,3 +57,5 @@
 
 	H.fully_replace_character_name(H.real_name, pick(GLOB.clown_names)) //rename the mob AFTER they're equipped so their ID gets updated properly.
 	H.dna.add_mutation(CLOWNMUT)
+	for(var/datum/mutation/human/clumsy/M in H.dna.mutations)
+		M.mutadone_proof = TRUE
```
Comment (Time-Green) [inline code/modules/jobs/job_types/clown.dm]: This is more like a band aid. Basically mutadone can remove all mutations, even ones created through stuff like wizardry. This is an oversight.

https://github.com/tgstation/tgstation/blob/e4e3ff3c49abc8836c82bb47e53f71528eea65c5/code/modules/reagents/chemistry/reagents/medicine_reagents.dm#L893
Pass the parameter list(MUT_NORMAL, MUT_EXTRA) in remove_all_mutations

Comment (Fikou) [reply code/modules/jobs/job_types/clown.dm]: M.dna.remove_all_mutations(MUT_NORMAL, MUT_EXTRA, mutadone = TRUE)
like this?

Comment (Time-Green) [reply code/modules/jobs/job_types/clown.dm]: M.dna.remove_all_mutations(list(MUT_NORMAL, MUT_EXTRA), TRUE)

Comment (Fikou) [reply code/modules/jobs/job_types/clown.dm]: ok done

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -890,7 +890,7 @@
 /datum/reagent/medicine/mutadone/on_mob_life(mob/living/carbon/M)
 	M.jitteriness = 0
 	if(M.has_dna())
-		M.dna.remove_all_mutations(mutadone = TRUE)
+		M.dna.remove_all_mutations(list(MUT_NORMAL, MUT_EXTRA), TRUE)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: why did u change the first arg

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: oh nvm

**Comment by ExcessiveUseOfCobblestone** [comment]
have your PR reflect all changes

**Comment by ATH1909** [comment]
Wait, are mutations with reinforcement chromosomes currently bugged and curable with mutadone? Does this PR fix that?

**Comment by Time-Green** [comment]
@ATH1909 they’re not.

**Comment by FloranOtten** [comment]
why DNM?

**Comment by ExcessiveUseOfCobblestone** [comment]
@Time-Green what does removing MUT_OTHER do?

**Comment by Time-Green** [comment]
@ExcessiveUseOfCobblestone MUT_OTHER removes all non-genetic mutations, like wizard mutate

**Pull request merged**